### ISSUE TITLE
White selection on dark themes when unfocussed

### ISSUE THREAD

**Issue by kanehekili** [tomboy-notes/tomboy-ng#291]
Tomboy-ng does not respect dark themes when it comes to selection. Background black, text white, die selection is blue, as it should (due to the underlying theme). As soon the note looses its focus, the selection area becomes white (which is a good contrast to the white text) 
The problem occurs on Linux Xfce with compositor.

**Comment by davidbannon** [comment]
Hi Kanehekii. Getting a reliable dark theme everywhere is just about impossible I have found. The problem on Linux is that I have no control over the Theme you are using. And some themes seem to make quite strange choices !  That combined with unpredictable DE rules about selection colours ...

Thats why I have now in tomboy-ng, the ability for you to select your own colours for everything in the Note Windows themselves. ( Sadly, I don't have that level of control in all the GUI elements.)

So, please try changing the colours you are using, Settings, Notes tab, lower right button "Set Colours".  It may help your problem but I cannot be sure it will given I cannot test with every DE and every Theme available for them.

If that does not find a more usable combination, please let me know which version of tomboy-ng, which distro and which theme you are using and I will see if I can do something about it.

Also, are you allowing tomboy-ng to read the theme or are you using the --dark-theme switch ?

Davo

**Comment by aguador** [comment]
Just tested on XFCE with Vertex Maia dark theme and its highlight/selection color, no --dark-theme switch and can confirm that when not focused the highlight becomes a very light grey, although the text (in pure white) can still be read.

Kanehikili, is that what is happening on your machine, or is the text no longer readable?

The same things happens on Enlightenment with the same theme, but custom highlight color. This does not happen with, e.g., L3afpad. That said, is functionality affected? In some ways as long as the selected text remains legible, it adds very easy to process confirmation that the window is not focused.

**Comment by kanehekili** [comment]
@davidbannon  Using the current Tomboy-ng Version from debian testing. As @aguador  mentioned, any dark theme will do. He describes the problem correctly. I will try to locate the problem, since I use my [own theme](https://www.gnome-look.org/p/1424967/)  and report back

**Comment by davidbannon** [comment]
OK, so, on my Debian Testing, XFCe, I find that without having a QT_QPA_PLATFORMTHEME set, tomboy-ng ignores the desktop setting, thats typical of a QT app I am afraid. Image 1. 

If you set QT_QPA_PLATFORMTHEME to, for example, qt5ct and have qt5ct set to "darker" theme, it does, as kanehekili says, shows selected text as white with a white background when the window looses focus. Image 2

 If you set qt5ct to 'dusk', then there is no problem, image number 3. But as a color scheme, its pretty terrible IMHO.

kanehekili, I have very little control of the colours in Qt5 apps, they pickup the Desktop Theme on some systems, on others you have to rely on tools like qt5ct to set the colors you like. As I mentioned, you can set the colours you want in tomboy-ng Settings->Notes but the Desktop Theme and Qt can and do overrule even some of those and it appears, that is the case here.

As far as I know, there is no way that I can override the "Selected Text" colour in either Qt or GTK apps. But we do get a lot more control over the colors used in GTK2, you may consider getting the GTK2 version instead of the Qt one. I only swapped to Qt because Debian are getting a bit pushy about GTK2. But I note they still include GTK2 libraries in the default install !

Its worth noting that the problem of white text on a white background also shows up when you are using the in-note search facility, Ctrl-F, tomboy-ng selects a bit of text for you when as it finds it but also sets that ugly white background. Its an issue with the theme I am afraid.  qt5ct lets you write your own theme, I am not sure how much control iy gives you however !

Davo 

![issue291-1](https://user-images.githubusercontent.com/6291248/222894166-9307e40c-682d-440a-87b9-4ffc062853e7.png)
![issue291-2](https://user-images.githubusercontent.com/6291248/222894180-11b02710-d30b-443b-9df9-cce444dfd68c.png)
![Issue291-3](https://user-images.githubusercontent.com/6291248/222894240-2fab39cc-f219-41d5-83eb-4e6526b20606.png)

**Comment by davidbannon** [comment]
One more thing, it seems to me (and I am not a dark theme user) that adding the --dark-theme to tomboy-ng's command line make it more usable. As you appear to have a QT_QPA_PLATFORMTHEME already set, you may need to disable it.

$> QT_QPA_PLATFORMTHEME=  tomboy-ng  --dark-theme 

The QT_QPA_PLATFORMTHEME setting does override it if allowed to ....

Davo

**Comment by davidbannon** [comment]
And one more thing again !
Its very easy to fix using qt5ct, image 4 shows tomboy-ng using the qt5ct "darker" mentioned above. But I copied it and set the 'Normal Text, Inactive' color to a light grey instead of vivid white, as shown in image 5

qt5ct does not seem to be installed by default on xfce, so its just a -

`$> sudo apt install qt5ct <enter>`

Start the tool itself with the command `qt5ct` and either set `QT_QPA_PLATFORMTHEME=qt5ct` env var in your shell scripts or just put it in front of any qt5 app you wish to adjust the colours with. qt5ct has a man page.

 Davo

![issue291-4](https://user-images.githubusercontent.com/6291248/222896180-c18624ac-aaa1-46e4-a07d-c42338686d76.png)
![issue291-5](https://user-images.githubusercontent.com/6291248/222896187-ee9dd4f8-d7e1-4db4-a8b2-df3193ec7b73.png)

**Comment by kanehekili** [comment]
Thanks Davo, your help (and work) is very much appreciated. You are right about the pickup of colors on different Desktops. I am using QT5/6 myself on XFCE, therefore I have usually installed the necessary libs like qt5ct or qt5-styleplugins. I'll test your suggestions and report back.

**Comment by davidbannon** [comment]
Hmm, i just realized you are probably using version 0.35.  V0.36 has just migrated from Sid to Testing this morning (my time). Does have a different look n feel, done, as much as anything, to make the buttons in the note window a bit more dark theme friendly.

Davo

**Comment by aguador** [comment]
Not to rain on anyone's parade, but my test was done with the Gtk2 version. So, while the Qt version may be "tricked",  the issue remains. I just tested the Gtk version with the --dark-theme switch: same result -- white text on a light grey background when not focused.

**Comment by davidbannon** [comment]
Yeah, good catch [aguador](https://github.com/aguador) - we can get around a bad theme on Qt5 (or Qt6) but not GTK2. But you should be able the change the (badly chosen) theme colours in -ng's colour setting facility.

Again, I say it, I cannot alter some decisions made by a theme. Best thing to do is choose a more sensible theme IMHO.

Davo

**Comment by kanehekili** [comment]
The new version V0.36 behaves the same on Debian testing, with the selection turning white on focus lost. In addition the urls are now black instead of blue, thus not really readable anymore:
![image](https://user-images.githubusercontent.com/4390452/223431868-c39a74df-79d4-4bde-8d23-8d7877da3b43.png)

**Comment by davidbannon** [comment]
But again kanehekili, its a theme problem, not a tomboy-ng one and one you can change using qt5ct.
Davo

**Comment by kanehekili** [comment]
I'm not sure if I can follow you. I've been writing serveral apps (see my repo here) with qt5 and don't have those problems.
I startet tomboy-ng with qt5c and dark mode:
![image](https://user-images.githubusercontent.com/4390452/223746940-986f8969-ab41-4afc-85cf-60256ebc38ec.png)
Neither buttons nor scrollbar respect the theme.

**Comment by davidbannon** [comment]
OK, so I built a new machine for this test, not a VM, a real machine based on the current weekly build of Testing. I built it with Gnome because its widely used and Mate because its my choice of a Desktop. As I expected, in neither case did tomboy-ng pickup its colours from the desktop. Sadly that is something we have lost with the switch to Qt5. I wonder if Qt6 will do better ? I am testing a prototype...

Four images shown, hopefully from the top -

- Gnome with var set to use qt5ct
- Gnome with --dark-theme switch
- Make with the var set to use qt5ct settings
- Mate with the --dark-theme switch.

As you can see, the Note window has some links showing and some are selected and then, another window is given focus. In every case everything is fully visible. I have not made any changes to the default colours here in either tomboy-ng nor qt5ct. (interesting I did not need to change the selected text colour as I had to a few days ago. I wonder if thats because I have used a very recent update and maybe Debian have fixed the problem settings ?).

**_Now, I just realized that I kanehekili uses XFCe ! I had him/her as a Gnome user. I'll try adding XFCe to that machine._**

Anyway, the fact is that different Themes use different colours. Thats a given, thats why people use different themes ! Sometimes those different colours don't work in a particular combination of desktop/theme/app_defaults. I cannot possibly test on every theme anyone was ever made.  

And I hate what the dark themes do to my button colors. I feel like giving up to be honest !
 
![Gnome_qt5ct_darker](https://user-images.githubusercontent.com/6291248/223947204-78d0fc36-0f5b-416b-b9c1-4c3684618bc3.png)
![Gnome_Switch](https://user-images.githubusercontent.com/6291248/223947231-640a7a99-c672-466a-b21e-be8fe5feacc9.png)
![Mate_qt5ct_Darker](https://user-images.githubusercontent.com/6291248/223947236-a41a6e3c-9311-4d5a-ac05-0206e56517a5.png)
![Mate_Switch](https://user-images.githubusercontent.com/6291248/223947242-c673f7bb-95b3-4c1d-920b-bb5d45b43dc4.png)

**Comment by davidbannon** [comment]
OK, virtually the same results in XFCe. Except, interesting, tomboy-ng picks up the dark theme from the desktop without having to use tricks like the --dark-theme switch or setting up qt5ct. So, that leaves me wondering what XFCe does that the others don't ?

I could not find Vertex Maia theme in my package manager, is it a standard XFCe theme ?

Anyway, I tried all the standard XFCe darker themes and got the same results as above. Links in that teal green, text in light grey. Selected Links and text in white, clearly visible.

Davo

**Comment by aguador** [comment]
> Yeah, good catch [aguador] - we can get around a bad theme on Qt5 (or Qt6) but not GTK2. But you should be able the change the (badly chosen) theme colours in -ng's colour setting facility.
> 
I have used Vertex themes for years and never seen problems with any of them, but have now tested with Adwaita (blindingly bright, ugh!), Adwaita Dark, and one of the Shades of Gray themes. The dark themes all behave the same, but there is a further problem. In all cases (including Adwiata) the note picked up the highlight color from the theme, not the one I had selected in -ng. I tested with changed background and text colors, and they were picked up, but not the highlight color.

These tests were with MX Linux, XFCE 4.18.1 and -ng 0.35a, Gtk2 version. Since this is not my primary OS and my Enlightenment systems use a highlight color similar to the default title color and the Vertex Maia colors, I had not noticed the problem on those systems. However, testing in Enlightenment showed the same issue: the highlight color was always inherited from the application theme, not ng.

I do believe that highlight colors worked with earlier releases, but I tested in XFCE with 0.31h and 0.33a (both Gtk2) and the highlight colors were NOT picked up.

I will note (and file as an issue if you like) that on MX Linux, XFCE 4.18.1, the menu text in 0.35a Gtk2 appears in bold. That does not happen with 0.35a in Enlightenment, nor in XFCE with earlier releases.

**Comment by kanehekili** [comment]
Testing the ng qt5 Version 0.36 with Adwaita-dark on xfce4 debian-testing : highlight color is taken from theme - which is fine and expected. On focus lost the selection becomes white - no matter which theme. Otherwise no complains, since I could render the text colors (except highlighting)  as Davo described it earlier. 

Concerning the buttons:
The QT5 Version pics up my gtk2 theme perfectly:
![image](https://user-images.githubusercontent.com/4390452/223990355-fc79f402-672c-4f2d-9652-d70901d5f210.png)

Theme used : https://www.pling.com/p/1424967/

**Comment by kanehekili** [comment]
The "white" hightlight also appears with the QT5 Version, when searching in a note. Looks like the colors setting of the selection does not work right. Check if the given color is valid. 

![image](https://user-images.githubusercontent.com/4390452/224024830-ccef9280-2c8d-47ce-87bd-63065e36b5a7.png)

**Comment by davidbannon** [comment]
OK, attached is a test binary, please git it a try. Its qt5 so must have the qt libraries available.  Just extract the binary from the zip ball into, perhaps, your home directory. Check that its still executable and that your existing tomboy-ng is NOT running. And then double click it. Try your tests ....

If you like it, I suggest you don't, just yet, put it in place of the real one, I will need to do a lot more tests before I am happy for you to use it (quite a lot of changes). For the record, I found two things -

- tomboy-ng Qt5 can be made to, sort of, follow the, qt5ct colours. So, bypassing my colours is a good idea. Most controls look heaps better here.
- It appears that KMemo, a component I use  for the Notes edit window, makes a few mistakes when using the qt5ct colours, thus the white on white for some colour themes. I have overridden one of those mistakes. Will chase TK up about it but not sure he will want to make an exception for Qt.

Lets see what you find !

[tomboy-ng.zip](https://github.com/tomboy-notes/tomboy-ng/files/10947893/tomboy-ng.zip)

**Comment by davidbannon** [comment]
Here is a further refined version. I have made extensive changes to the whole color system. If you use qt5ct, you get only the colors from the qt5ct theme. I have tested it with the themes provided by qt5ct and it is usable (but not great). Nothing more I can do !

I will be very interested in your test results. I would also like to see a screen shot of the colours from your qt5ct with this scheme you use.
[Uploading tomboy-ng.zip…]()
 Open qt5ct, click the button ('...') to right of the color scheme combo, choose Edit. (if Edit is not offered in menu, you need to copy the existing scheme, select the copy, and try again).

Davo

**Comment by kanehekili** [comment]
The first version displays the colors correctly (I do not have qt5ct installed, my env is QT_QPA_PLATFORMTHEME=gtk2). Selection on an exiting note works correctly:
![image](https://user-images.githubusercontent.com/4390452/224947522-9b764607-7a74-42ea-9f94-51fbbe891573.png)

However a **new Note** behaves like before:
![image](https://user-images.githubusercontent.com/4390452/224948229-ebe887ab-1a9b-40d4-8264-4c01d36332a7.png)

The second version behaves differently concerning the highlighting: (since not qt5ct is set) (Selection is black :-( )
![image](https://user-images.githubusercontent.com/4390452/224949640-83004f77-0d67-466b-8955-f9656824e800.png)

No change of color if unfocussed. I'd prefer the first version

**Comment by davidbannon** [comment]
Hmm, please go back and test either (or both) **without** the QT_QPA_PLATFORMTHEME=gtk2 

The reason I ask is that I understand you are using Debian Testing (?) and it does not install the gtk2 themes by default so you should find that the env var makes no difference. Unless you have installed them manually of course. Because of that omission, I decided to ignore the gtk2 setting. Debian is now treating any GTK2 dependency as a bug, a minor bug but a bug nevertheless.

The gtk2 setting works fine on my U20.04 box but on Testing (Bookworm) just gives the default colors, hard wired into the app before it starts to look at where it is running.

David

**Comment by kanehekili** [comment]
Setting QT_QPA_PLATFORMTHEME="" (since it is a global) results in a default non dark color-theme:
![image](https://user-images.githubusercontent.com/4390452/224959564-8d5962df-17ff-4091-9de8-e8d26bb4ba3a.png)

**Comment by davidbannon** [comment]
Yeah, of course. Thats was silly of me. The fact you are getting any dark theme indicates you have 'a' gtk2 theme installed.

OK, so that means you have installed 'some' gtk2 themes ?  I am pretty sure Vertex Maia theme does not come from Debian, maybe I am wrong ? I cannot replicate your problem on any of the standard themes (but I have not tried XFce, should I ?).

On my Debian Testing test boxes, all attempts to get a qt5 app to use the 'out of the box' themes fail. 'gtk2', gtk3', 'gnome' all revert back to my initial hard wired colors. To get a dark theme on Debain Testing, I must use qt5ct and choose one of its dark themes.

And sorry, I won't be shipping the March 11 version, it contains a couple of hard wired colors that will mess up a lot of people. I am starting to think we need to go back the tomboy-ng's built in color chooser, because you are not using QT_QPA_PLATFORMTHEME=qt5ct, you should be able to select your own colors from in there.

Just as an aside, I would question the wisdom of basing your project on a theme that apparently differs in its color use from the 'mainstream' ones. I test against "out of the box" color schemes and even that is getting far to complicated.  Sigh...

David

**Comment by kanehekili** [comment]
I'd like to point out that I am the creator of a theme which I referenced [here](https://github.com/tomboy-notes/tomboy-ng/issues/291#issuecomment-1461726123). I never used Vertex Maia and will never use it, (Even though I'm an avid Manjaro user) I still expect that themes have a gtk2 support. Because gtk2 is the only way to have GTK3/4 themes and QT Themes look alike. If GTK2 is dropped completely, I might be in trouble. So you have a valid point.
To avoid confusion:; I'm using tomboy-ng on debian-testing(work) and the classical tomboy on Arch/Manjaro(home). On all systems, I use my own theme (obviously) which supports GTK2. Using QT software like VBox shows that QT and GTK themes start to diverge - you need fusion to get a half decent look and feel (==ugly)

**Comment by davidbannon** [comment]
OK, so I should be a bit more careful in what I say !  Sorry.  I understood you were using qt5ct to manage colors and have been concentrating on getting it to work sensibly with all of the themes that come with qt5ct. Ignoring, to some degree, the others.

In actual fact, with Qt5 trying to look like GTK2 (color wise) is actually easy. Qt5 apps do display the GTK2 theme fairly faithfully except for KMemo, the component that I use for the note editor. It sets a pretty reasonable light theme and stops there. So, in my code I look at other components in the app, if the background on (from memory) the settings window is dark, I override the KMemo's colors with dark ones of my choice.

Thats where the problem crops up, I am not a dark theme user, working with a dark theme gives me a head ache so, you guessed it, I do not do a good job !  (On the other hand, the light color schemes look pretty good ;-) ).

Later tonight, I will revise the dark colors I superimpose onto KMemo when there is not some other set of colors to be choosen (ie qt5ct colors). 

Have some other things I must do now ....

Davo

**Comment by davidbannon** [comment]
OK [kanehekili](https://github.com/kanehekili), referring to your test of the tomboy-ng.zip files that I uploaded. You really have me struggling here I am afraid. Can you please do a fresh download of the second  tomboy-ng.zip, the one that will report ('About' from main menu) itself as being dated the 13th March.

I have downloaded that one myself and it does not look anything like how you reported it yesterday. Both the version in my code here now and (to be sure) the one I downloaded, dated 13th, produce a jet black background and a grey background on selection. 

I am wondering if -

1. You have selected different colors in the tomboy-ng selection tool, Settings->Notes->Colors.  If that is the case, those selections will override gtk2. Just open that selection tool and click "default".
2. You have mixed up the versions ?

Note that, as I understand it, everyone using a dark theme should see it the same, it does NOT use the dark colors you have carefully selected, it just detects what it thinks is a dark theme and uses a set of colors I determine. Possibly badly. Personally, I hate the jet black but will be interested to hear your view as a regular Dark Theme user. Unlike working from eg qt5ct, the colors we are talking about now are very easy for me to change, make a suggestion ! 

Here is how I see the one dated 13th.

![291](https://user-images.githubusercontent.com/6291248/225251252-5af00aae-6cd4-4958-a86a-3024d5285523.png)

Davo

**Comment by davidbannon** [comment]
OK, as I mentioned, I don't like the jet black, here is a slighly softer version.
![issue291-7](https://user-images.githubusercontent.com/6291248/225293803-f962c0cb-0462-4c90-888b-2ff350412691.png)

**Comment by aguador** [comment]
Just a quick apology to the two of your for muddying the waters with my mention of the theme whose name I will not repeat here. I really just joined in this thread because I saw that the issue exists for the gtk2 version as well (which Davo has confirmed as a KMemo issue). Logical given that [kanehekili](https://github.com/kanehekili) has: QT_QPA_PLATFORMTHEME=gtk2.

Davo, Your "softer" colors are generally better. The dark application themes I know do not use jet black.

Off-topic note to [kanehekili](https://github.com/kanehekili):  Nice theme - glad you left "materiality" to Madonna.:-) The last two screenshots from Pling seem to be Cinnamon from the settings window and mint green folders. Nothing is highlighted in those 2 screenshots. Is there a non-blue highlight color available?

**Comment by davidbannon** [comment]
Your comments always welcome Roy. I feel like applying what I have learnt in this discussion with a new 'release' to Debian if its not too late (must be very close). Even I can see it looks better than then my earlier attempts. But I need to work out how  [kanehekili](https://github.com/kanehekili) got such a different result than I do using the 13th March one.

Just to be sure, here is yet another zip, tomboy-ng-16.zip, it has that newer, softer background, brighter text I mentioned yesterday. It describes itself as being made on the 16th.  This nicer colors will not be seen by most Debian Bookworm users of course, they won't have GTK2. But some might. Any feedback welcome.

[tomboy-ng-16.zip](https://github.com/tomboy-notes/tomboy-ng/files/10985049/tomboy-ng-16.zip)

**Comment by kanehekili** [comment]
> But I need to work out how [kanehekili](https://github.com/kanehekili) got such a different result than I do using the 13th March one.

Not sure. I've installed qt5ct, set the style to "gtk2" and exported the QT_QPA_PLATFORMTHEME=qt5ct. Using "fusion" renders the widgets distinctly different from the native theme. 
The advantage of using gtk2 for the platformtheme is that the colors and look & feel are close to the selected GTK theme.

Calling tomboy-ng from Mar.13 with export QT_QPA_PLATFORMTHEME=qt5c seems to ignore any theming:
![image](https://user-images.githubusercontent.com/4390452/225638224-5b88c21d-9b8b-41b2-a343-648b73ec1098.png)

I have the same problem with my own QT-Apps with qt5ct: The colors are not respected. It looks like the settings are ignored.

environment:
`QT_ACCESSIBILITY=1
QT_QPA_PLATFORMTHEME=qt5c
QT_STYLE_OVERRIDE=fusion
`
Switching back to gtk2 resolves the issue

**Comment by davidbannon** [comment]
OK just a few things to clarify (and sorry for the long message).

I have not tested using qt5ct pointing back to gtk2. Thats an additional level of complication in an already complicated discussion. Maybe that qt5ct gtk2 setting is about dialogs rather colour ?  If you want gtk2 for this test, then please just use 
`>$ QT_QPA_PLATFORMTHEME=gtk2  ./tomboy-ng`

(Its possible that qt5ct can read the actual GTK2 them colors and make them available to my app. So, worth looking at but only after we understand the basic things. Like why you are not seeing a grey selection background.)

Next, with the exception of `QT_QPA_PLATFORMTHEME=qt5ct`, tomboy-ng does not attempt to follow any particular theme colours in the note editing screen. It uses a list of colors of my devising (that I would like you to comment on), either 'light' or 'dark'. So, we get the 'dark' version because - '--dark-theme' given on command line or the app detects other components in the app are being shown dark. So, no matter what your theme says, if its dark, you get my dark colours, if its light, you get my light colours. Its a case of getting those pre selected colours right. The rest of the app does follow the theme in some cases.
 
[kanehekili](https://github.com/kanehekili), you seem to be setting `QT_QPA_PLATFORMTHEME=qt5c` I believe its actually `qt5ct`, not `qt5c`.

kanehekili - you seem to be testing with the 11th March version. That is not a viable release candidate, it includes a number of 'hard wired' colors as a test of the colour scheme it self, not the application. Someone using, for example, a light theme would still see those dark, hard wired colors.

The test I would really appreciate is to run the binary in the tomboy-ng-16.zip file with your gtk2 dark theme active gtk2 -
`>$ QT_QPA_PLATFORMTHEME=gtk2  ./tomboy-ng`

Please dispose of the previous test binaries. I cannot replicate their colors nor, more importantly, the rules to display which colour, their code has been overwritten. 

Note also, a command line switch '--dark-theme' overrides everything, it is  a last resort, does not do anything to most the non kmemo widgets.  Next, the user selected colours take precedence, if set, those colours are used no what others might be more appropriate. Click 'default' in the colour selection window to take it out of equation.

Sorry this has become so complicated !

Davo

**Comment by davidbannon** [comment]
[aguador](https://github.com/aguador) - there should be absolutely no reason to start a new session. You do have to restart tomboy-ng after a theme change, thats all.  Do make sure you do not have another instance of tomboy-ng running, if you do, the existing one is activated and the new one exits.

As I mentioned above, unless you pass control over to qt5ct, the kmemo component (which provides the editor) uses one of two sets of predefined colours, 'light' and 'dark'. The rest of the app may follow theme, but kmemo does not.

> A question: Given that ng cannot control the highlight colour...

Well, I can control the highlight and selection background colour in the KMemo. And do so, in the example I hope you are using, its set to gray. I can change it but need to understand why K sees black. I 'believe' that nothing you or he can do will alter it but he sees black !

Davo

**Comment by kanehekili** [comment]
`export QT_QPA_PLATFORMTHEME=qt5ct`  (pointing to gtk2)
`bin/tomboy-tests/tomboy16-ng`

works out. 
The selection in the note is gray - not the theme selection color, but readable. Focus lost behaviour ok, no white block:
![image](https://user-images.githubusercontent.com/4390452/225846361-50a6613f-2182-4451-86ba-1bfdd1f82773.png)

`export QT_QPA_PLATFORMTHEME=gtk2`

works out.
The selection in the note is now black. Focus lost behaviour ok:
![image](https://user-images.githubusercontent.com/4390452/225847663-7ccc2a30-1628-4963-bf78-d9af0f225756.png)


`export QT_QPA_PLATFORMTHEME=qt5ct`  (pointing to fusion)

From my point of view not acceptable since the theme does not match any more (Wrong list selection color, wrong button gradient etc) but no real color break:

![image](https://user-images.githubusercontent.com/4390452/225848631-59e578e0-6604-404b-824c-328c82160cad.png)

**Comment by davidbannon** [comment]
> export QT_QPA_PLATFORMTHEME=gtk2 works out. The selection in the note is now **black**. Focus lost behaviour ok:

I simply don't know what to say. There is absolutely no where in my code what I set the selection background to be black. And I do set the KMemo colours as I explained. You must be hitting this bit of code -

```
	if DarkTheme or DarkThemeSwitch then begin
            BackGndColour:= $303030;        // KMemo Background
            AltColour  := $606060;          // Some panel's background color
            HiColour   := $600001;          // a dark blue;  This is, eg Crtl H type highlighting, not selection !
            TextColour := clWhite;
            TitleColour:= $B8B800;
            LinkColour := $B8B801;
            AltBackGndColor := clGray;     // Selected text, both focused and unfocused
    end else begin                                     
```
and this -
```
   KMemo1.Colors.SelBkGnd := Sett.AltBackGndColor;         // Selected backgnd when looses focus
   KMemo1.Colors.SelBkGndFocused := Sett.AltBackGndColor;  // Selected backgnd with focus       
```
I cannot replicate what you see with any possible combo. And I have tried a lot of combos !

**_I have a Bookworm XFCe box now, please tell me what themes I need to install and use to replicate your system._**


> export QT_QPA_PLATFORMTHEME=qt5ct (pointing to fusion). From my point of view not acceptable since the theme does not match any more (Wrong list selection color, wrong button gradient etc) but no real color break:

[kanehekili](https://github.com/kanehekili), I have absolutely no control of the list selection colour (etc) in this case. Its come from qt5ct, just were qt5ct gets it from is a mystery to me because you have not mentioned what color scheme is set in qt5ct. But what ever it is, it has nothing to do with me. I manually set the kmemo colours but not things like list selection (unless you are using --dark-theme switch).

> Back to Version March 11, export QT_QPA_PLATFORMTHEME=gtk2, Here the selection-color fits the theme. So if you could add this tweak to your March 16 Version it'll be perfect.

Again, I cannot set things like list selection colour, its set, in this case from gtk2.

If you see different results in the list selection colour, button color, gradients between -

```
$> QT_QPA_PLATFORMTHEME=gtk2   ./tomboy-ng 
$> QT_QPA_PLATFORMTHEME=qt5ct  ./tomboy-ng
```
(where qt5ct is set to display gtk2) Then its a case of qt5ct not faithfully passing the gtk2 data to the app. Can you see that ? My sticky fingers are not involved in any way.

Davo

**Comment by aguador** [comment]
Davo: I DK what the problem was with the black, but I did not have other instances of ng running. If I get a moment later, may play with the qt5ct option in MX Linux (bullseye base).

[kanehekili](https://github.com/kanehekili): OT - Not too fussy about colors, but sick of blue (although your darker Adwaita-like blue is better than others). Obviously I like the Vertex Maia color scheme with nselected_bg_color:#16a085,; Shades of Grey Patina green: "selected_bg_color: #3c7460"; or Shades of Grey Harvest: "selected_bg_color: #AD8641". Even the Shades of Grey Firebrick can be an interesting choice: "selected_bg_color: #c14e4e"

**Comment by davidbannon** [comment]
OK, this will probably amuse you two.
I just tested the current code base on Windows. And I got a BLACK background to selected text (with Black text of course). I was not amused.
So, 3 hours later, six cycles of booting into windows, testing, shutting down, waiting for windows updates, booting into Linux, starting Virtual Box and the VM I do my cross compile from, edit, compile.....

I found that, as I mentioned to both of you several times but I forgot, having colours set in the colour picker overrides everything else. Now, apart from providing some laughs, this is interesting. The tomboy-ng built in colour chooser does NOT set the selection colour but it sure was in Windows. How ? I really don't know. 

I will, for a start, add a 'hint' to the colour chooser button telling anyone that custom colours are in use. 

But I cannot find any pathway as to how it affected the selected background colour, I have to wonder if KMemo if being smarter than I give it credit for ? Trying to set an appropriate selection background colour for a given normal background ?

Davo

**Comment by kanehekili** [comment]
@davidbannon  I don't laugh, since that color stuff is not trivial.  As I said your version from March 11 suites my needs and supports gtk2 and the GTK(3&4) theme selection color..  Since I haven't touched Windows for 15 years I'd admit that this is not in my focus.  I think the main problem is using  "KMemo", being QT centric. Using more primitive Widgets like QTextEdit would probably solve your problem.

**Comment by davidbannon** [comment]
[kanehekili](https://github.com/kanehekili), I have wrapped it up as it is now and sent if off to Debian, they are in "Hard Freeze" already so I'll have to work to get it accepted. Should not be too hard as it not a dependency of any other package as far as I know.  I will get the normal release out, via github, in a day or so, I like to release them all together and there is quite an array of platforms.

I agree with you, Linux is my sole using platform but Windows uses make up more than half of my (github) downloads. But thats not allowing for PPA and direct from the repo Linux users.

Thanks for you help with this problem [kanehekili](https://github.com/kanehekili), it was way, way past time I paid more attention to dark theme users in particular. I still think there might be a potential problem if users use the custom colour tool within tomboy-ng but I feel I have mitigated it and, probably, will remove the DIY colour model, its not needed now we know so much more about Qt5 colours.

Davo
